### PR TITLE
refactor: examples/delegated-routing

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR refactors the `delegated-routing` example to the refactored async `js-libp2p`.

### CODE REVIEW

File: examples/delegated-routing/package.json [hunk 1]
```diff
@@ -3,19 +3,18 @@
...
-    "libp2p-kad-dht": "~0.14.12",
-    "libp2p-mplex": "~0.8.5",
-    "libp2p-secio": "~0.11.1",
-    "libp2p-webrtc-star": "~0.15.8",
-    "libp2p-websocket-star": "~0.10.2",
-    "libp2p-websockets": "~0.12.2",
-    "react": "^16.8.6",
-    "react-dom": "^16.8.6",
-    "react-scripts": "2.1.8"
+    "ipfs": "~0.40.0",
```
Comment (vasco-santos) [inline examples/delegated-routing/package.json]: We need to get `libp2p` to `js-ipfs` before getting this to work. 

However, I could also not test this with #master . Do we need to run `go-ipfs` in the specific branch mentioned in the README?

Comment (jacobheun) [reply examples/delegated-routing/package.json]: There are some configuration issues that need to be resolved, see https://github.com/libp2p/js-libp2p/pull/371#pullrequestreview-255991314. 

In regards to requiring `ipfs`, we may want to revisit this example and only have libp2p in the client. The example cats a file right now, but in reality we just need to find things on the network via the delegate node.

Comment (vasco-santos) [reply examples/delegated-routing/package.json]: For this example, should we have a `js-libp2p` node running and providing a few cid that we can then find with the app example?

File: examples/delegated-routing/.env [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# required because react-scripts scans *up* the tree from this project and finds
+# a conflicting version of eslint in the node_modules dir for js-ipfs.
```
Comment (achingbrain) [inline examples/delegated-routing/.env:2]: Is this necessary?  js-ipfs doesn't have a runtime dependency on eslint?  Maybe if this example was in the js-ipfs repo?

Comment (vasco-santos) [reply examples/delegated-routing/.env:2]: This is needed. I copied this from the react example of `js-ipfs`, but did not read the comment: https://github.com/ipfs/js-ipfs/blob/master/examples/browser-create-react-app/.env
It is aegir that installs a version of `webpack` that is not compatible with `react-scripts`. I am going to change that comment

**Comment by vasco-santos** [comment]
After syncing with @jacobheun , we will start by having a public delegated router configured.

We were having some issues finding peer via the delegate because of the lack of support for HTTPS in the `go-ipfs` API. This needs to be properly configured using a reverse proxy and generated SSL certificates. I will create an issue to track adding documentation for infrastructure for going to production to unblock this PR for now.

**Comment by achingbrain** [comment]
Superseded by #1390

**Pull request closed without merging**